base/bucket_gocb_test.go

```diff
@@ -1062,6 +1064,73 @@ func CouchbaseTestDeleteDocumentHavingXATTR(t *testing.T) {
 
 }
 
+// TestDeleteDocumentUpdateXATTR.  Delete document that has a system xattr along with an xattr update.
+func CouchbaseTestDeleteDocumentUpdateXattr(t *testing.T) {
+	b := GetBucketOrPanic()
+	bucket, ok := b.(*CouchbaseBucketGoCB)
+	if !ok {
+		log.Printf("Can't cast to bucket")
+		return
+	}
+
+	// Create document with XATTR
+	xattrName := "_sync"
+	val := make(map[string]interface{})
+	val["body_field"] = "1234"
+
+	xattrVal := make(map[string]interface{})
+	xattrVal["seq"] = 1
+	xattrVal["rev"] = "1-1234"
+
+	key := "TestDeleteDocumentHavingXATTR"
+	_, _, err := bucket.GetRaw(key)
+	if err == nil {
+		log.Printf("Key should not exist yet, expected error but got nil.  Doing cleanup, assuming couchbase bucket testing")
+		bucket.Delete(key)
+	}
+
+	// Create w/ XATTR, delete doc and XATTR, retrieve doc (expect fail), retrieve XATTR (expect success)
+	cas := uint64(0)
+	cas, err = bucket.WriteCasWithXattr(key, xattrName, 0, cas, val, xattrVal)
+	if err != nil {
+		t.Errorf("Error doing WriteCasWithXattr: %+v", err)
+	}
+
+	// Delete the document.
+	err = bucket.Delete(key)
+	if err != nil {
+		t.Errorf("Error doing Delete: %+v", err)
+	}
+
+	// Verify delete of body was successful, retrieve XATTR
+	var retrievedVal map[string]interface{}
+	var retrievedXattr map[string]interface{}
+	getCas, err := bucket.GetWithXattr(key, xattrName, &retrievedVal, &retrievedXattr)
+	if err != nil {
+		t.Errorf("Error doing GetWithXattr: %+v", err)
+	}
+	assert.Equals(t, len(retrievedVal), 0)
+	assert.Equals(t, retrievedXattr["seq"], float64(1))
+	log.Printf("Post-delete xattr (1): %s", retrievedXattr)
+	log.Printf("Post-delete cas (1): %x", getCas)
+
+	// Update the xattr only
+	xattrVal["seq"] = 2
+	xattrVal["rev"] = "1-1234"
+	casOut, writeErr := bucket.WriteCasWithXattr(key, xattrName, 0, getCas, nil, xattrVal)
+	assertNoError(t, writeErr, "Error updating xattr post-delete")
+	log.Printf("WriteCasWithXattr cas: %d", casOut)
+
+	// Retrieve the document, validate cas values
+	var postDeleteVal map[string]interface{}
+	var postDeleteXattr map[string]interface{}
+	getCas2, err := bucket.GetWithXattr(key, xattrName, &postDeleteVal, &postDeleteXattr)
+	assertNoError(t, err, "Error getting document post-delete")
+	log.Printf("Post-delete xattr (2): %s", postDeleteXattr)
+	log.Printf("Post-delete cas (2): %x", getCas2)
```

Comment:
Could there be additional assertions on what's expected in `postDeleteXattr` and `postDeleteVal`?  (and I guess `getCas2` as well?)